sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala

```diff
@@ -106,21 +106,27 @@ private[hive] class HiveClientImpl(
 
     // Set up kerberos credentials for UserGroupInformation.loginUser within
     // current class loader
-    // Instead of using the spark conf of the current spark context, a new
-    // instance of SparkConf is needed for the original value of spark.yarn.keytab
-    // and spark.yarn.principal set in SparkSubmit, as yarn.Client resets the
-    // keytab configuration for the link name in distributed cache
-    if (sparkConf.contains("spark.yarn.principal") && sparkConf.contains("spark.yarn.keytab")) {
+    if (sparkConf.contains("spark.yarn.principal")) {
```

Comment:
One sort of hacky way is to have `ApplicationMaster.scala` override the value of `KEYTAB` in the configuration with the value of `AM_KEYTAB`. Then this code just needs to look at "spark.yarn.keytab" to cover both cases.